code/game/objects/structures/window.dm

```diff
@@ -345,8 +345,9 @@
 /obj/structure/window/on_update_icon()
 	//A little cludge here, since I don't know how it will work with slim windows. Most likely VERY wrong.
 	//this way it will only update full-tile ones
-	color =  material.icon_colour
-	if(reinf_material)
+	if(istype(material))
```

Comment:
As discussed earlier I am completely against putting `istype()` checks into code that should not be executing prior to init, just to avoid your code throwing runtimes downstream when you do weird shit.

Reply:
The code under if() wouldn't work if it wasn't of istype(). Regardless of its relation to my downstream, the if check as coded is incorrect when applied without context.

Reply:
It isn't incorrect to assume that a variable is holding the correct value of its type when it's validated elsewhere. That is the point of having typed variables in the first place.

Reply:
Except typed variables are type abused by the original authors of this code.